python/src/scipp/compat/mantid.py

```diff
@@ -771,6 +771,16 @@ def convert_TableWorkspace_to_dataset(ws, error_connection=None, **ignored):
     return dataset
 
 
+def convert_WorkspaceGroup_to_dataarray_dict(group_workspace, **kwargs):
+    workspace_dict = {}
+    for i in range(group_workspace.getNumberOfEntries()):
+        workspace = group_workspace.getItem(i)
+        workspace_name = workspace.name().replace(group_workspace.name(), '')
```

Comment:
I think this is OK. i.e. the group key naming is independent of ADS key naming. If not the case then this is the potential source of bugs.

Reply:
I don't think the GroupWorkspace and it's contents can ever have identical names as they exist in a Flat structure in the ADS under the hood.